### PR TITLE
add dot and l2_normalize operators

### PR DESCRIPTION
**Pull request by simanw304**
add dot and l2_normalize operators
fix predict (https://github.com/intel-analytics/zoo/issues/315)

### CODE REVIEW

File: pyzoo/test/zoo/pipeline/autograd/test_operator.py [hunk 1]
```diff
@@ -32,6 +32,8 @@
 class TestOperator(ZooTestCase):
     # shape including batch
     def compare_binary_op(self, kk_func, z_layer, shape):
```
Comment (zhichao-li) [inline pyzoo/test/zoo/pipeline/autograd/test_operator.py]: compare_binary_op(self, kk_func, z_layer, shape, rtol = 1e-5, atol = 1e-5)

File: pyzoo/zoo/pipeline/api/autograd.py [hunk 2]
```diff
@@ -50,6 +50,29 @@ def abs(x):
     return Variable.from_jvalue(callBigDlFunc("float", "abs", x))
 
 
+def dot(x, y, axes=1, normalize=False):
```
Comment (zhichao-li) [inline pyzoo/zoo/pipeline/api/autograd.py]: Rename to `batch_dot` and enrich the description.

File: pyzoo/zoo/pipeline/api/autograd.py [hunk 3]
```diff
@@ -195,6 +218,10 @@ def softplus(x):
     return Variable.from_jvalue(callBigDlFunc("float", "softplus", x))
 
 
+def mm(x, y, axes):
```
Comment (zhichao-li) [inline pyzoo/zoo/pipeline/api/autograd.py]: add doc

File: pyzoo/zoo/pipeline/api/keras/engine/topology.py [hunk 4]
```diff
@@ -191,15 +191,15 @@ def predict(self, x, distributed=True):
                      Default is True. In local mode, x must be a Numpy array.
         """
         if distributed:
-            if isinstance(x, np.ndarray):
+            if isinstance(x, np.ndarray) or isinstance(x, list):
```
Comment (zhichao-li) [inline pyzoo/zoo/pipeline/api/keras/engine/topology.py]: Revert this change as `to_sample_rdd` might not support list of ndarray for now.

File: zoo/src/main/scala/com/intel/analytics/zoo/pipeline/api/autograd/math.scala [hunk 5]
```diff
@@ -175,6 +175,50 @@ object AutoGrad {
       bnn.Contiguous[T]().asInstanceOf[AbstractModule[Activity, Activity, T]]).inputs(input.node)
     Variable(contiguousNode)
   }
+
+  def mm[T: ClassTag](
```
Comment (zhichao-li) [inline zoo/src/main/scala/com/intel/analytics/zoo/pipeline/api/autograd/math.scala]: add doc

**Comment by zhichao-li** [comment]
Update `math.md` accordingly.

**Comment by simanw304** [comment]
jenkins passed http://172.168.2.101:8080/view/ZOO-PR/job/ZOO-PR-Validation/685/
